runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -900,4 +939,38 @@ private String capsString(String str)
 	{
 		return str.substring(0, 1).toUpperCase() + str.substring(1);
 	}
+
+	// check if player is carrying the item
+	private boolean itemOnPlayer(int itemID)
+	{
+		return checkItemContainer(InventoryID.INVENTORY, itemID) || checkItemContainer(InventoryID.EQUIPMENT, itemID);
+	}
+
+	private boolean checkItemContainer(InventoryID id, int itemID)
+	{
+		final ItemContainer itemContainer = client.getItemContainer(id);
+		// can item be substituted with slayer helmet?
+		final boolean useSlayerHelm = Task.SLAYER_HELMET_ITEMS.contains(itemID);
+
+		if (itemContainer == null)
+		{
+			return false;
+		}
+
+		final Item[] container = itemContainer.getItems();
```

Comment:
Instead of iterating the container's items, [use `ItemContainer#contains()`](http://static.runelite.net/api/runelite-api/net/runelite/api/ItemContainer.html#contains(int))